darts/models/forecasting/tft_model.py

```diff
@@ -0,0 +1,836 @@
+            },
+            'model_config': {}
+        }
+
+        reals_input = []
+        time_varying_encoder_input = []
+        time_varying_decoder_input = []
+        static_input = []
+        for input_var in type_names:
+            if input_var in variables_meta['input']:
+                vars = variables_meta['input'][input_var]
+                reals_input += vars
+                if input_var in ['past_target', 'past_covariate', 'historic_future_covariate']:
+                    time_varying_encoder_input += vars
+                elif input_var in ['future_covariate']:
+                    time_varying_decoder_input += vars
+                elif input_var in ['static_covariate']:
+                    static_input += vars
+
+        variables_meta['model_config']['reals_input'] = list(dict.fromkeys(reals_input))
+        variables_meta['model_config']['time_varying_encoder_input'] = list(dict.fromkeys(time_varying_encoder_input))
+        variables_meta['model_config']['time_varying_decoder_input'] = list(dict.fromkeys(time_varying_decoder_input))
+        variables_meta['model_config']['static_input'] = list(dict.fromkeys(static_input))
+
+        return _TFTModule(
+            variables_meta=variables_meta,
+            output_dim=self.output_dim,
+            input_chunk_length=self.input_chunk_length,
+            output_chunk_length=self.output_chunk_length,
+            hidden_size=self.hidden_size,
+            lstm_layers=self.lstm_layers,
+            dropout=self.dropout,
+            num_attention_heads=self.num_attention_heads,
+            hidden_continuous_size=self.hidden_continuous_size,
+            likelihood=self.likelihood
+        )
+
+    def _build_train_dataset(self,
+                             target: Sequence[TimeSeries],
+                             past_covariates: Optional[Sequence[TimeSeries]],
+                             future_covariates: Optional[Sequence[TimeSeries]]) -> MixedCovariatesSequentialDataset:
+
+        return MixedCovariatesSequentialDataset(target_series=target,
+                                                past_covariates=past_covariates,
+                                                future_covariates=future_covariates,
+                                                input_chunk_length=self.input_chunk_length,
+                                                output_chunk_length=self.output_chunk_length,
+                                                max_samples_per_ts=self.max_sample_per_ts)
+
+    def _verify_train_dataset_type(self, train_dataset: TrainingDataset):
+        raise_if_not(isinstance(train_dataset, MixedCovariatesTrainingDataset),
+                     'TFTModel requires a training dataset of type MixedCovariatesTrainingDataset.')
+
+    def _produce_train_output(self, input_batch: Tuple):
+        return self.model(input_batch)
+
+    def predict(self, n, *args, **kwargs):
```

Comment:
If we leave `n`, and `n < self.output_chunk_length`, the inference dataset only loads the next `n` future covariates. 
So to get a complete output chunk for prediction, we would have to step back 
`-(self.output_chunk_length - n)` and then do prediction for `n_new = self.output_chunk_length`.
This means that we ignore the last values from `past_target`.

To avoid this, I intercepted calls to `predict()` so that we load a minimum of `self.output_chunk_length` points from future data.